algebras/algebra/src/test/scala/endpoints/algebra/client/EndpointsTestSuite.scala

```diff
@@ -233,6 +233,47 @@ trait EndpointsTestSuite[T <: EndpointsTestApi] extends ClientTestBase[T] {
         encodeUrl(path / evenNumber) (42) shouldEqual "/42"
       }
 
+      "xmap query string" should {
+        import client._
+
+        "xmap locations" in {
+          case class Location(longitude: Double, latitude: Double)
+
+          val locationQueryString: QueryString[Location] =
+            (qs[Double]("lon") & qs[Double]("lat"))
+              .xmap(
+                { case (lon, lat) => Location(lon, lat) },
+                location => (location.longitude, location.latitude))
+          
+          encodeUrl(path /? locationQueryString) (Location(12.0, 32.9)) shouldEqual "?lon=12.0&lat=32.9"
+          encodeUrl(path /? locationQueryString) (Location(-12.0, 32.9)) shouldEqual "?lon=-12.0&lat=32.9"
+          encodeUrl(path /? locationQueryString) (Location(Math.PI, -32.9)) shouldEqual s"?lon=${Math.PI}&lat=-32.9"
+        }
+
+        "xmapPartial blogids" in {
+          sealed trait BlogId
+          case class BlogUuid(uuid: UUID) extends BlogId
+          case class BlogSlug(slug: String) extends BlogId
+
+          val blogIdQueryString: QueryString[BlogId] =
+            (qs[Option[UUID]]("uuid") & qs[Option[String]]("slug"))
+              .xmapPartial {
+                case (maybeUuid, maybeSlug) =>
+                  maybeUuid.map[BlogId](BlogUuid).orElse(maybeSlug.map[BlogId](BlogSlug))
+              } {
+                case BlogUuid(uuid) => (Some(uuid), None)
+                case BlogSlug(slug) => (None, Some(slug))
+              }
+          
+          val testUUID: String = "f4b9defa-1ad8-453f-9a06-2683b8564b8d"
```

Comment:
Here too, you probably want to work with a value of type `UUID` from the beginning.